src/styles/getTooltipStyles.js

```diff
@@ -121,6 +121,8 @@ export default function getTooltipStyles(
     // Determine if we need to shift the tooltip horizontally to prevent it
     // from displaying too near to the edge of the screen.
     const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth, manualShiftHorizontal);
+    const horizontalShiftPointer = horizontalShift > 0 ? Math.max(-horizontalShift, -(tooltipWidth / 2) + (POINTER_WIDTH / 2) + variables.componentBorderRadiusSmall)
+        : Math.min(-horizontalShift, (tooltipWidth / 2) - (POINTER_WIDTH / 2) - variables.componentBorderRadiusSmall);
```

Comment:
In my proposal, I fix the case component is positioned on the extreme right by replace -horizontalShift by `Math.min(-horizontalShift, (tooltipWidth / 2) - (POINTER_WIDTH / 2) - variables.componentBorderRadiusSmall)`, and it could be fixed similarly for the case extreme left. So when horizontalShift > 0 that is case extreme left and vice versa

Reply:
@fedirjh Thanks. I just update some comment above it, please check again.

Reply:
@shawnborton In the PR, I not only decrease `gutterWidth` but also calculate `horizontalShiftPointer` to prevent in the furture has the component is position over safe area, the arrow will move slightly as your suggestion.